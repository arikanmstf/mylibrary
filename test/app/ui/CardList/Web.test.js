/**
 * Web Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';

const props = {
  cards: [{
    title: 'dummyTitle',
    description: 'dummyDescription',
    image: 'dummyImage',
    id: 1,
    isFavorite: true,
    isRead: true,
  }],
  isLoaderVisible: false,
  search: 'dummySearch',
  addCards: jest.fn(),
  fetchPublications: jest.fn(),
};

describe('test/app/ui/CardList/Web.test.js', () => {
  let wrapper;

  beforeAll(() => {
    window.addEventListener = jest.fn();
    window.removeEventListener = jest.fn();
    const { CardList } = require('ui/CardList/Web'); // eslint-disable-line global-require
    wrapper = shallow(<CardList {...props} />);
  });

  afterAll(() => {
    window.addEventListener.clearMock();
    window.removeEventListener.clearMock();
  });

  it('Render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
