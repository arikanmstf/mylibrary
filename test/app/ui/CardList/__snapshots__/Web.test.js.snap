// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/CardList/Web.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    addCards={[MockFunction]}
    cards={
      Array [
        Object {
          "description": "dummyDescription",
          "id": 1,
          "image": "dummyImage",
          "isFavorite": true,
          "isRead": true,
          "title": "dummyTitle",
        },
      ]
    }
    fetchPublications={[MockFunction]}
    isLoaderVisible={false}
    search="dummySearch"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Connect(withRouter(CardDetail))
          card={
            Object {
              "description": "dummyDescription",
              "id": 1,
              "image": "dummyImage",
              "isFavorite": true,
              "isRead": true,
              "title": "dummyTitle",
            }
          }
          style={
            Object {
              "margin": "10px 0",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "card": Object {
            "description": "dummyDescription",
            "id": 1,
            "image": "dummyImage",
            "isFavorite": true,
            "isRead": true,
            "title": "dummyTitle",
          },
          "style": Object {
            "margin": "10px 0",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Connect(withRouter(CardDetail))
            card={
              Object {
                "description": "dummyDescription",
                "id": 1,
                "image": "dummyImage",
                "isFavorite": true,
                "isRead": true,
                "title": "dummyTitle",
              }
            }
            style={
              Object {
                "margin": "10px 0",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "card": Object {
              "description": "dummyDescription",
              "id": 1,
              "image": "dummyImage",
              "isFavorite": true,
              "isRead": true,
              "title": "dummyTitle",
            },
            "style": Object {
              "margin": "10px 0",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
